### PR TITLE
PYTHON-3256 Obtain AWS credentials for CSFLE in the same way as for MONGODB-AWS

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: bindings/python/pymongocrypt/mongocrypt.py [hunk 1]
```diff
@@ -35,7 +35,8 @@
 class MongoCryptOptions(object):
     def __init__(self, kms_providers, schema_map=None, encrypted_fields_map=None,
                  bypass_query_analysis=False, crypt_shared_lib_path=None,
-                 crypt_shared_lib_required=False, bypass_encryption=False):
+                 crypt_shared_lib_required=False, bypass_encryption=False,
+                 use_need_kms_credentials_state=False):
```
[the remark this answers is not in the document]
Comment (blink1073) [reply bindings/python/pymongocrypt/mongocrypt.py:39]: How about `enable_on_demand_kms_creds`?

File: bindings/python/test/test_mongocrypt.py [hunk 2]
```diff
@@ -595,6 +620,7 @@ def test_rewrap_many_data_key(self):
         raw_doc = RawBSONDocument(result)
         assert len(raw_doc['v']) == 2
 
+
```
[the remark this answers is not in the document]
Comment (blink1073) [reply bindings/python/test/test_mongocrypt.py:623]: Removed, one day we'll run `black` here.  :smile:

File: bindings/python/test/test_mongocrypt.py [hunk 3]
```diff
@@ -347,6 +350,7 @@ def __init__(self,
         self.key_docs = key_docs
         self.kms_reply = kms_reply
         self.kms_endpoint = None
+        self.got_on_demand_credentials = False
```
[the remark this answers is not in the document]
Comment (blink1073) [reply bindings/python/test/test_mongocrypt.py:353]: Added a comment

File: bindings/python/pymongocrypt/mongocrypt.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply bindings/python/pymongocrypt/mongocrypt.py:39]: Do we need to add a new option at all? Can we implemented the provide_kms_providers() callback in pymongocrypt instead?

[the remark this answers is not in the document]
Comment (blink1073) [reply bindings/python/pymongocrypt/mongocrypt.py:39]: We could move the logic here and still tell the user to install `pymongo[aws]` if `pymongo_auth_aws` isn't available, that sounds good.

**Pull request merged**